code-name-taran/src/com/spaghettisoft/component/chess/gamelogic/GameLogic.java

```diff
@@ -0,0 +1,906 @@
+		
+	    int[] directions = getDirection(move);
+		int directionRow = directions[ZERO];
+		int directionColum = directions[ONE];
+
+		
+		do {
+				currentRow += directionRow;
+				currentColumn += directionColum;
+
+				if (!IsRowAndColumAreMatrixRange(currentRow, currentColumn) || ((currentRow == move.getToRow()) && (currentColumn == move.getToColumn())) ) {
+					break;
+				}
+				 figure = this.chessBoard.getFigure(currentRow, currentColumn);
+			if(figure != null) {
+				result = true;
+				break;
+			}
+			
+		} while (true);
+		return result;
+	}
+	
+	private boolean isFieldInCheck(Player player, int row, int column) {
+		int counter = 0; 			
+		AbstractFigure figure;
+		int[] figureCordinates;
+		Move move;
+
+		for (int i = 0; i < this.chessBoard.getBoardLenght(); i++) {
+			for (int j = 0; j < this.chessBoard.getBoardLenght(); j++) {
+				if (isThisFigureKnight(i, j) && isEnemyFigure(player, i, j)) {
+					if(this.chessBoard.getFigure(i, j).validMove(new Move(i, j, row, column))) {
+						return true;
+					}
+				}
+			}
+		}
+	
+		while (counter < directions.length) {
+			figureCordinates = getPositionOfFirstFoundFigureInGivenDirection(player, row, column, directions[counter][ZERO], directions[counter][ONE]);
+
+			if (figureCordinates != null) {
+				figure = this.chessBoard.getFigure(figureCordinates[ZERO],figureCordinates[ONE]);
+				move = new Move(figureCordinates[ZERO],figureCordinates[ONE], row, column);
+				if ( isEnemyFigure(player, figureCordinates[ZERO], figureCordinates[ONE]) && figure.validMove(move) ){
+					if( !(isThisFigurePawn(figureCordinates[ZERO], figureCordinates[ONE]) && (figureCordinates[ONE] == column)) ) {
+						return true;
+					} 
+				}
+			}
+			 counter++;
+		}
+		return false;
+	}
+	
+	
+
+
+	private int[] getPositionOfFirstFoundFigureInGivenDirection(Player player, int row, int column, int rowDirection, int columnDirection) {
+		int currentRow = row;
+		int currentColumn = column;
+		int[] figureCordinates = null;
+		AbstractFigure figure;
+		
+		while (true) {
+				currentRow += rowDirection;
+				currentColumn += columnDirection;
+				
+				if (!IsRowAndColumAreMatrixRange(currentRow, currentColumn) ) {
+					break;
+				}
+				figure = this.chessBoard.getFigure(currentRow, currentColumn);
+				if ( figure!= null && !(!isEnemyFigure(player, currentRow, currentColumn) && isThisFigureKing(figure)) ) {
+					figureCordinates = new int[TWO];
+					figureCordinates[ZERO] = currentRow;
+					figureCordinates[ONE] = currentColumn;
+					return figureCordinates;
+			}
+		}
+		return figureCordinates;
+	}
+
+	private boolean IsRowAndColumAreMatrixRange(int currentRow, int currentColum) {
+		boolean result = (ZERO <= currentRow && currentRow < this.chessBoard.getBoardLenght()) && 
+				      	 (ZERO <= currentColum && currentColum < this.chessBoard.getBoardLenght());
+		return result;
+	}
+	
+	private boolean isKignInCheckWhenMoveFigure(Player player, Move move) {

```

Comment:
The method is way too large. You should try to split such methods in several sub-methods.